### PR TITLE
feat: deprecateOption

### PR DESCRIPTION
**Pull request by laggingreflex**
Shows a `[deprecated]` notice in front of an option

* Added a method: `.deprecateOption(key, [message])`
* Added a key in option conrtructor: `.option(key, { deprecated: [boolean|message] })`
* Basic Tests
* Updated Readme

Work towards #1518

### CODE REVIEW

**Comment by mleguen** [comment]
Thank you @laggingreflex 

I will try to review this ASAP, but I am afraid I won't have time to do it before next week.

File: lib/usage.js [hunk 1]
```diff
@@ -294,6 +295,9 @@ module.exports = function usage (yargs, y18n) {
         if (~options.number.indexOf(key)) type = `[${__('number')}]`
 
         const extra = [
+          (key in deprecatedOptions)
+            ? `[${__('deprecated')}${typeof deprecatedOptions[key] === 'string'
```
Comment (mleguen) [inline lib/usage.js]: Could you please add the new translation(s) to `locales/en.json`?

File: lib/usage.js [hunk 2]
```diff
@@ -294,6 +295,9 @@ module.exports = function usage (yargs, y18n) {
...
+              ? `: ${deprecatedOptions[key]}` : ''}]` : null,
```
[the remark this answers is not in the document]
Comment (laggingreflex) [reply lib/usage.js]: fixed

File: lib/usage.js (hunk 1 above)
Comment (laggingreflex) [reply lib/usage.js]: done

File: test/usage.js [hunk 3]
```diff
@@ -313,6 +313,47 @@ describe('usage tests', () => {
...
+      it('{deprecated: string} should show [deprecated: string]', () => {
+        const r = checkUsage(() => yargs('--help')
+          .option('x', { deprecated: 'string' })
+          .wrap(null)
+          .parse()
+        )
+        r.logs[0].should.include('  -x  [deprecated: string]')
+      })
+    })
+    describe('using .deprecateOption(x, [string])', () => {
```
Comment (bcoe) [inline test/usage.js]: this option should apply `globally` right, could we add a test that a deprecated command set at the top level carries to sub-commands?

Comment (laggingreflex) [reply test/usage.js]: done

**Comment by bcoe** [comment]
@laggingreflex thanks for the contribution :+1:

**Comment by bcoe** [comment]
@laggingreflex mind trying `npm i yargs@next`, and making sure your feature works as expected?

Thanks again for the contribution.

**Comment by laggingreflex** [comment]
Yep, working as expected.

**Comment by friederbluemle** [comment]
Nice feature! Any chance this can be included in the `@types/yargs` TypeScript definitions as well?

**Comment by mleguen** [comment]
@friederbluemle In the `@types/yargs` definitions, I am not sure, but it the typescript definitions yargs should soon ship directly, yes.

**Comment by tamj0rd2** [comment]
Has this been released? I haven't been able to use this in a typescript project using yargs 15.3.1

**Pull request merged**